src/utils/AuthProvider.tsx

```diff
@@ -22,7 +22,7 @@ export interface AuthContextType {
     email: string,
     password: string,
     options: object,
-  ) => Promise<AuthResponse>;
+  ) => Promise<{ name: string; message: string } | null>;
```

Comment:
is there a reason why we decided to transition away from `AuthResponse`?

we might not need it for sign up, but it could ultimately benefit the future dev team if this sign up wrapper provides the same API return values as a regular Supabase sign up function.

Reply:
@pragyakallanagoudar correct me if I'm wrong, but I it seems like the type was updated to fit the solution starting on line 108 taken from https://github.com/orgs/supabase/discussions/1282#discussioncomment-5230475. Could we update the code so that we go back to using `AuthResponse`? This is the structure of an `AuthResponse`, we should be able to directly update its error message:

<img width="219" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/afa2f9c9-38fb-4059-b19b-01802e4adf97">
<img width="329" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/a77ad0e8-0233-4802-ad40-fd5268ee7898">